src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala

```diff
@@ -0,0 +1,286 @@
+package ai.privado.languageEngine.javascript.tagger.collection
+
+import ai.privado.cache.RuleCache
+import ai.privado.model.{Constants, InternalTag, RuleInfo}
+import ai.privado.tagger.PrivadoParallelCpgPass
+import ai.privado.utility.Utilities._
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.{Block, Call, Method, MethodRef}
+import io.shiftleft.semanticcpg.language._
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+import scala.util.{Failure, Success, Try}
+
+class CollectionTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCpgPass[RuleInfo](cpg) {
+  private val logger       = LoggerFactory.getLogger(this.getClass)
+  private val methodUrlMap = mutable.HashMap[Long, String]()
+  private val classUrlMap  = mutable.HashMap[Long, String]()
+
+  override def generateParts(): Array[RuleInfo] =
+    ruleCache.getRule.collections.filter(_.catLevelTwo == Constants.default).toArray
+
+  override def runOnPart(builder: DiffGraphBuilder, collectionRuleInfo: RuleInfo): Unit = {
+    val collectionMethodsCache = mutable.ListBuffer[Method]()
+
+    // TODO: Add this one to Internal APIs
+    // Added this for now to log the endpoints, in future need to move client side endpoint to Internal API
+//    val CLIENT_ENDPOINT_PATTERN  = "(?:axios|fetch|express|@angular/common/http).*"
+//    val apiClientCollectionCalls = cpg.call.methodFullName(CLIENT_ENDPOINT_PATTERN).l
+//    for (call <- apiClientCollectionCalls) {
+//      if (call.argument.nonEmpty) {
+//        val isValid = getCollectionMethodsCache(call, call.method.id())
+//        if (isValid) {
+//          collectionMethodsCache += call.method
+//        }
+//      }
+//    }
+
+    // Supporting below pattern
+    // fastify.get('/endpoint', async (request, reply) => {
+    //  return 'Hello, World!';
+    // });
+    // Supported Framework: Express, Fastify, Featherjs
+    // TODO: Based on below frameworks improve the logic
+    // TODO: Need to support more frameworks Hapijs, Koa, Loopback, Sails, Restify, Connect, AdonisJS
```

Comment:
@dbMundada in this case well lets create a ticket for adding handling for the frameworks mentioned here. TODO: helps to address logical issues or tech debts. For features lets create a ticket.